Modules/Loadable/Markups/Widgets/qSlicerMarkupsPlaceWidget.cxx

```diff
@@ -180,6 +180,25 @@ vtkMRMLMarkupsFiducialNode* qSlicerMarkupsPlaceWidget::currentMarkupsFiducialNod
   return d->CurrentMarkupsNode;
 }
 
+//-----------------------------------------------------------------------------
+vtkMRMLInteractionNode* qSlicerMarkupsPlaceWidget::interactionNode()const
+{
+  Q_D(const qSlicerMarkupsPlaceWidget);
+  return d->InteractionNode;
+}
+
+//-----------------------------------------------------------------------------
+void qSlicerMarkupsPlaceWidget::setInteractionNode(vtkMRMLInteractionNode* interactionNode)
+{
+  Q_D(qSlicerMarkupsPlaceWidget);
+  if (d->InteractionNode == interactionNode)
+    {
+    return;
+    }
+  this->qvtkReconnect(d->InteractionNode, interactionNode, vtkCommand::ModifiedEvent, this, SLOT(updateWidget()));
```

Comment:
We may need to call `setInteractionNode(NULL)` in the destructor to avoid memory leaks (there may be indirect mechanisms that unobserver the interaction node, so maybe it explicit reset in destructor may not be necessary).

Reply:
Since we use a weak pointer, there is need to  be explicit. 

Regarding the observation, the connection is removed when the widget is deleted. See https://github.com/commontk/CTK/blob/6cd82aab5f21db7c021e1eb502c9e5161e4a636d/Libs/Visualization/VTK/Core/ctkVTKConnection.cpp#L468-L473

That said, it is better to be explicit. :+1: